metatron-app/metatron-discovery
# F#2352 enable prep spark engine

### Description
Now we can use Apache Spark for data preparation.

**Related Issue** : https://github.com/metatron-app/metatron-discovery/issues/2352


### How Has This Been Tested?
1. Download discovery spark engine: https://github.com/metatron-app/discovery-spark-engine
2. mvn -DskipTests clean install
3. Put configuration like below:
- polaris:
  - dataprep:
    - etl:
      - spark:
        - jar: `<discovery-prep-spark-engine-1.2.0.jar path>`
        - port: `<port>`

(example)
- polaris:
  - dataprep:
    - etl:
      - spark:
        - jar: `$HOME/git-repos/discovery-spark-engine/discovery-prep-spark-engine/target/discovery-prep-spark-engine-1.2.0.jar`
        - port: `5300`

3. ./run-prep-spark-engine.sh <configuration file path>
  - If not given, $METATRON/conf/application-config.yaml is used.
  - ex> ./run-prep-spark-engine.sh $METATRON_HOME/discovery-server/src/main/resources/application.yaml
4. Run discovery server as usual.
5. Import & wrangle [s5k_1.csv.txt](https://github.com/metatron-app/metatron-discovery/files/3514188/s5k_1.csv.txt)
6. Click "Snapshot"
7. Open "Advanced options"
8. Click "Spark" option.
9. Click "Done"
10. Check the result snapshot.

NOTICE:
1. Although, the jar location is on sprint framework's property file, it's parsed by bash shell script.
This is for unified configuration of metatron programs.
Therefore, ${user.home} is not right in this case. $HOME will work.
2. Discovery spark engine do not need any type of legacy spark cluster. Just we need is HCatalog. HCatalog is configured by _polaris.storage.stagedb.metastore.uri_

#### Need additional checks?
Yes, please.

### Types of changes



### Checklist:
